ludwig/utils/defaults.py

```diff
@@ -223,7 +223,7 @@ def update_feature_from_defaults(config: Dict[str, Any], feature_dict: Dict[str,
         default_encoder_or_decoder = get_default_encoder_or_decoder(feature_dict, config_feature_group)
         if default_params_for_feature_type[TYPE] != default_encoder_or_decoder:
             # Update type and populate defaults for the encoder or decoder type
-            feature_dict[parameter] = default_params_for_feature_type[TYPE]
```

Comment:
@connor-mccorm Are you able to try the test in this PR? https://github.com/ludwig-ai/ludwig/pull/2342

The new thing this is testing is if the encoder's additional dependencies (pretrained path, for e.g.) get injected into the input feature config. I have a feeling this will fail with the current implementation of this code block because with the new refactor, the `TYPE` check on line 221 will not be hit and it'll directly try line 233. The issue with that is the line 223 - 227 actually get the encoder or decoder class and then pull it any additional default parameters that are needed for that encoder or decoder class. The fix would just involve doing this step incase there's an encoder type specified in `default_params_for_feature_type`. 

Let me know if this makes sense!

Reply:
Will merge and fix this up. Thanks for the additional test!

Reply:
Just pulled, looks like it's passing with a couple of changes to the indexing logic.

Reply:
@connor-mccorm that's awesome!